### DISCUSSION TITLE
Wanhao D7+, Chitubox, and other conundrums

### DISCUSSION THREAD

**Discussion by lslewis901** [sn4k3/UVtools#182] [Q&A] [answered]
Hello, So I have questions about a couple of things in UVTools.  

First of all let me start by saying that the Wanhao d7 firmware requires certain things in a CWS file - and I have not figured out from the settings in UVTools what those are.  My first problem is any repaired layer/resin/island issues where I have created a CWS file using UVTools will invariably split the image on a single layer into 4 separate squashed regions rather than one continuous layer image.  

So I get 4 separate blobs building up rather than the actual object.

Secondly - the Chitubox plugin for creating a CWS file starts off during the save by naming of image files with a 001 rather than a 000 or 0000.  Being a developer (AKA code monkey) I saw where the UVTools code internals always assume that things should start at zero.  Which makes good sense sense to me and probably the Chitubox CWS file would never print anyway on the Wanhao firmware.

So the only two apps I have ever gotten to work slicing STL files for the Wanhao are Creation Workshop and LycheeSlicer.

But even then if I take one of those CWS files and run it thru UVTools the image per layer appears to get split into 4 parts across the x axis of the LED display.

Any suggestions, thoughts or ideas or anything???

Thanks

**Comment by X3msnake** [comment]
Image is probably being saved as Nova3D format, the DXQ6" screens uses that
encoding to work

What are your steps when opening and saving your sliced file on UVTools?

Lindsey Lewis ***@***.***> escreveu no dia sexta, 9/04/2021
à(s) 00:46:

> Hello, So I have questions about a couple of things in UVTools.
>
> First of all let me start by saying that the Wanhao d7 firmware requires
> certain things in a CWS file - and I have not figured out from the settings
> in UVTools what those are. My first problem is any repaired
> layer/resin/island issues where I have created a CWS file using UVTools
> will invariably split the image on a single layer into 4 separate squashed
> regions rather than one continuous layer image.
>
> So I get 4 separate blobs building up rather than the actual object.
>
> Secondly - the Chitubox plugin for creating a CWS file starts off during
> the save by naming of image files with a 001 rather than a 000 or 0000.
> Being a developer (AKA code monkey) I saw where the UVTools code internals
> always assume that things should start at zero. Which makes good sense
> sense to me and probably the Chitubox CWS file would never print anyway on
> the Wanhao firmware.
>
> So the only two apps I have ever gotten to work slicing STL files for the
> Wanhao are Creation Workshop and LycheeSlicer.
>
> But even then if I take one of those CWS files and run it thru UVTools the
> image per layer appears to get split into 4 parts across the x axis of the
> LED display.
>
> Any suggestions, thoughts or ideas or anything???
>
> Thanks
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sn4k3/UVtools/discussions/182>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACUR56T2SSCUWZXCFN337WTTHY56PANCNFSM42T4WTXQ>
> .
>


-- 
Com os melhores cumprimentos,
Vinicius Silva

**Reply by lslewis901** [reply]
Hello, I typically use the File Open, and once that happens I get a message about mismatched resolutions.  

> "Layer image resolution of {Width=2560, Height=1440} mismatch with printer resolution of {Width=0, Height=0}.
> Printing this file can lead to problems or malformed model, please verify your slicer printer settings;
> Processing this file with some of the tools can lead to program crash or misfunction;
> If you used PrusaSlicer to slice this file, you must use it with compatible UVtools printer profiles (Help - Install profiles into PrusaSlicer)."

So I just found the change the print resolution setting.  Thats probably it.  Now I feel stupid.

Thanks much... :)

**Reply by sn4k3** [reply]
That happens because the file you are loading lack of resolution information, ie: slicer not outputing that variable. You can send me your file so i can check whatever key they use for resoltuion
I can put a auto fixer for that situations...

**Comment by sn4k3** [comment, marked as answer]
I've made the .xml.cws format based on a slicer i don't renember which (Creation Workshop or the other), but my recommendation is to dump all that and use the PrusaSlicer with my printer profile. Have you tried? 
As i don't own the printer i can only improve things when someone open issues and start debuging the problems.
Post some images of the problem, and if possible the sliced file.

For me it looks ok when converting to Creation workshop file
![image](https://user-images.githubusercontent.com/113281/114112810-0ee0c680-98d5-11eb-92a6-b857323fb5ac.png)

**Reply by lslewis901** [reply]
Hello,  I tried using Prusa Slicer, but seemed to run into some issues following the exact steps to import the profile so I did not go back to that yet.  I will give it a go later today.

Thanks,
Lindsey

**Reply by sn4k3** [reply]
https://github.com/sn4k3/UVtools/wiki/Setup-PrusaSlicer

**Comment by lslewis901** [comment]
Hello,

I've attached 3 sample CWS files:

1. dice_Creation_Workshop.cws from CreationWorkshop v75
2. Chitubox_CWSplugin_dice.cws from Chitubox using the 32 bit plugin found here (https://github.com/JNJ3D/CWS-plugin-32bit/blob/master/cws_plugin_32bit.CHplugin)
3. Lychee_dice.cws from the Lychee slicer v3.xx

Interestingly the first from Creation Workshop opened fine without any error messages.  So I cannot recall if I tried printing from the UVTools saved CWS file after editing when it came from Creation Workshop.

Chitubox version of course generates the index out of range error, which I monkeyed around and got to open okay by making UVTool code modifications.  But It still did not print, probably because of 2 reasons - missing resolution settings and because print parameters were not correct for resin I was using.  

NOTE: I had tried using a different author's version of a Chitubox plugin (https://github.com/Zloyduh/cws), whose C++ source code I found on github, but it does not actually seem to generate the cws file.  I was thinking I could try to fix it.  But have not had the time.

Last from Lychee slicer I got the 4 blobs to print after setting correct print parameters, but did not realize resolution settings were missing despite the error message.  My bad.


[Sample CWS Files.zip](https://github.com/sn4k3/UVtools/files/6287804/Sample.CWS.Files.zip)

**Reply by sn4k3** [reply]
Chitubox_CWSplugin_dice: Have two gcode files, dont know why...
dice_Creation_Workshop: Looks to work fine, in the end my format was made taking the Creation Workshop way of doing the file
Lychee_dice.cws: Lacks of resolution paramenter, but again it is not required to print and that's way is ok to miss that. In next version of UVtools i will auto fix resolution if empty.

So my advice is either use PrusaSlicer with my profiles OR Lychee and manual fix resolution or wait for new patch

**Reply by sn4k3** [reply]
If you want to use Chitubox dump all plugins, export to ZIP, open in UVtools and convert to Creation Workshop, that easier than try to use messed up plugins

**Reply by lslewis901** [reply]
I will give PrusaSlicer a more robust try.  I simply ran into a missed step importing the profile and quit.  I use PrusaSlicer almost exclusively for FDM printing.

**Reply by lslewis901** [reply]
Thanks for your support.  Excellent tool.  I only found it because of Lychee.
